MarkUsProject/Markus
# add lti launch methods

## Motivation and Context


Adds the routes required for launching MarkUs via LTI, as well as allowing an LTI deployment to be associated with a markus course.

## Your Changes


**Description**:
Adds launch, redirect_login, and choose_course methods on LTI controllers, a basic model for lti deployments, and modifies the login method to redirect to choose_course route if launched from LTI and no course has been associated with a given Lti object.

Also adds new dependency [jwt](https://github.com/jwt/ruby-jwt) for encoding/decoding JWTs, and [webMock](https://github.com/bblimke/webmock) dependency for mocking web requests.

**Type of change** (select all that apply):







## Testing


Tested via launching markus from canvas and logging in as an instructor, as well as a student (though students shouldn't see the "launch markus" button). Added tests for new methods, and updated main_controller tests.

Instructions for testing from the local UI:
- If you haven't already, follow all testing steps listed in #5999 
- add `host.docker.internal` to `known_jwks` (LtiController line 92)
- update `iss` param in LtiController line 108 to have the value ` "https://canvas.instructure.com"`
- Click "Launch MarkUs" from within the canvas course where markus was installed

## Questions and Comments (if applicable)




Also TODO: add errors via rails_performance

## Checklist

- [x] I have reviewed the test coverage changes reported on Coveralls.
